powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/exception/DefaultExceptionResolver.java

```diff
@@ -132,6 +110,7 @@
     @ExceptionHandler(InvalidOperationContextException.class)
     @ResponseStatus(HttpStatus.BAD_REQUEST)
     public @ResponseBody ErrorResponse handleInvalidOperationContextException(InvalidOperationContextException ex) {
+        logger.error("Invalid operation context", ex);
```

Comment:
I would use `warn` level to log everything which is not caused by an application error. This is user's bad request data.

Reply:
Although I agree with the warning level, this change would introduce an unecessary difference in our reference and in client implementations. So I prefer to keep it as it is for now. If you feel strongly about it feel free to raise an issue, but we will have to synchronize this change with client implementation to avoid uncessary diffs in our code bases.